test/test-ciphers-from-pod.sh

```diff
@@ -0,0 +1,58 @@
+#!/bin/bash
+
+# $SERVER is expected to be provided as an environment variable
+# and should represent the location of the console via internal
+# IP address, represented by pod.status.podIP, and including :8443.
+# example: 10.128.0.30:8443
+echo "connecting to ${SERVER}"
+
+
+# ensure we ignore weak ciphers
+# CBC (cipher block chaining) are no longer reliable and should not be used
+# CBC ciphers use an IV (initialization vector) and a chaining mechanism.
+# The chaining mechanism means that a single bit error in a ciphertext block
+# will invalidate all previous blocks.  The chaining was good in that it hides
+# plaintext patterns, but is inferior to other cipher modes.
+INVALID_CIPHER_SAMPLE=(
+  DES-CBC3-SHA                  # from QE
+  RSA-AES-128-CBC-SHA256
+  ECDHE-RSA-3DES-EDE-CBC-SHA    # disabled to mitigate SWEET32 attack
+  RSA-3DES-EDE-CBC-SHA          # disabled to mitigate SWEET32 attack
+)
+
+echo "checking for console"
+curl --head "${SERVER}"
+
+for CIPHER in "${INVALID_CIPHER_SAMPLE[@]}"
+do
+  RESULT=$(openssl s_client -connect "${SERVER}" -cipher "${CIPHER}" 2>&1)
+  # note that this swallows errors if openssl doesn't exist in the container
+  if [[ $? -eq 0 ]]
```

Comment:
This is bad, I used to check for specific strings in the output, which might be better.  This will swallow errors if openssl isn't installed.  On the flip side, with the `VALID_CIPHER_SAMPLE` tests below, these also fail.  The reason is just unclear (the kind of reasons that suck in CI when you forget past pains).